var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -111,189 +107,77 @@ class FoamExtend(Package):
     depends_on('scotch~metis+mpi', when='+ptscotch')
     depends_on('metis@5:',         when='+metis')
     depends_on('parmetis',         when='+parmetis')
-    depends_on('parmgridgen',      when='+parmgridgen')
+    # mgridgen is statically linked
+    depends_on('parmgridgen',      when='+parmgridgen', type='build')
     depends_on('paraview@:5.0.1',  when='+paraview')
 
-    # Some user settings, to be adjusted manually or via variants
-    foam_cfg = {
-        'WM_COMPILER':        'Gcc',  # <- %compiler
-        'WM_ARCH_OPTION':     '64',   # (32/64-bit on x86_64)
-        # FUTURE? 'WM_LABEL_SIZE':      '32',  # <- +int64
-        'WM_PRECISION_OPTION': 'DP',  # <- +float32
-        'WM_COMPILE_OPTION':  'SPACKOpt',   # Do not change
-        'WM_MPLIB':           'USER',       # USER | USERMPI
-    }
+    # General patches
+    common = ['spack-Allwmake', 'README-spack']
+    assets = []
 
-    # The system description is frequently needed
-    foam_sys = {
-        'WM_ARCH':      None,
-        'WM_COMPILER':  None,
-        'WM_OPTIONS':   None,
+    # Some user config settings
+    config = {
+        'label-size': False,    # <- No int32/int64 support
+        'mplib': 'USERMPI',     # USER | USERMPI
     }
 
+    # The openfoam architecture, compiler information etc
+    _foam_arch = None
+
     # Content for etc/prefs.{csh,sh}
     etc_prefs = {}
 
     # Content for etc/config.{csh,sh}/ files
     etc_config = {}
 
-    build_script = './spack-Allwmake'  # <- Generated by patch() method.
-    # phases = ['configure', 'build', 'install']
-    # build_system_class = 'OpenfoamCom'
+    phases = ['configure', 'build', 'install']
+    build_script = './spack-Allwmake'  # <- Added by patch() method.
+
+    #
+    # - End of definitions / setup -
+    #
 
     def setup_environment(self, spack_env, run_env):
-        run_env.set('FOAM_INST_DIR', self.prefix)
+        run_env.set('FOAM_INST_DIR', os.path.dirname(self.projectdir)),
+        run_env.set('FOAM_PROJECT_DIR', self.projectdir)
         run_env.set('WM_PROJECT_DIR', self.projectdir)
-
-    @property
-    def _canonical(self):
-        """Canonical name for this package and version"""
-        return 'foam-extend-{0}'.format(self.version.up_to(2))
+        for d in ['wmake', self.archbin]:  # bin already added automatically
+            run_env.prepend_path('PATH', join_path(self.projectdir, d))
+        run_env.set('MPI_BUFFER_SIZE', "20000000")
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        """Provide location of the OpenFOAM project.
+        This is identical to the WM_PROJECT_DIR value, but we avoid that
+        variable since it would mask the normal OpenFOAM cleanup of
+        previous versions.
+        """
+        spack_env.set('FOAM_PROJECT_DIR', self.projectdir)
 
     @property
     def projectdir(self):
```

Comment:
@olesenm Makes sense. I can only imagine how difficult it would be to make such a substantial change to the build system, especially now that OpenFOAM development has been split into 3 different branches: openfoam-com, openfoam-org, and foam-extend. I assume you primarily develop openfoam-com?

Also, thank you so much for taking the time to get these packages to build. I remember the last time one of our users requested OpenFOAM (org), I spent several weeks just trying to get it to build. I spent another several weeks trying to package it into Spack based on the foam-extend package, but had no luck. It's really nice when developers package their own software because I imagine you know what you're doing a lot better than I do.